cmd/kubeadm/app/phases/kubelet/kubelet.go

```diff
@@ -51,6 +56,10 @@ func CreateBaseKubeletConfiguration(cfg *kubeadmapi.MasterConfiguration, client
 		return err
 	}
 
+	if err := writeInitKubeletConfigToDisk(kubeletBytes); err != nil {
```

Comment:
This won't work. You need to write down this config RIGHT AFTER the controlplane/etcd phases in order for the kubelet to even start. So please extract this logic into a function you can call in `init.go` after the controlplane stuff, but before you start waiting for the API to be available